### ISSUE TITLE
Add Regression losses

### ISSUE THREAD

**Issue by 5uperpalo** [jrzaurin/pytorch-widedeep#59]
Add losses that show promising results for regression prediction in scenarios with highly imbalanced datasets used e.g. in Life Time Value prediction
* ZILN loss paper https://github.com/google/lifetime_value/blob/master/lifetime_value/zero_inflated_lognormal.py
  * loss and prediction method https://github.com/google/lifetime_value/blob/master/lifetime_value/zero_inflated_lognormal.py
* Quantile loss 
  * https://pytorch-forecasting.readthedocs.io/en/latest/api/pytorch_forecasting.metrics.QuantileLoss.html
* Tweedie loss - generalized Poisson loss(?)
  * https://discuss.pytorch.org/t/custom-tweedie-loss-throwing-an-error-in-pytorch/76349/7
  * nice short summary with tweedie loss implementations https://github.com/awslabs/gluon-ts/issues/984#issuecomment-734420556

**Comment by 5uperpalo** [comment]
additional notes:
**tweedie loss**
* I took the initial idea and implementation from this [post](https://towardsdatascience.com/tweedie-loss-function-for-right-skewed-data-2c5ca470678f) and related [paper](https://arxiv.org/pdf/1811.10192.pdf)
* almost similar implementation [here](https://discuss.pytorch.org/t/custom-tweedie-loss-throwing-an-error-in-pytorch/76349/6)
* my brain is starting to hurt after spending 2 days figuring out following:
  * (i) let's say you have predictions that follow certain distribution, eg. e-commerce that follows some type of Tweedie distribution; (ii) let's say you have definition of this distribution, e.g. Tweedie; how the hell do you transform distribution into loss function?
    * I understand there exists [distribution deviance](https://en.wikipedia.org/wiki/Deviance_(statistics)), ie "goodness-of-fit"
    * you may find these goodnesses-of-fit in [scikit](
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_tweedie_deviance.html)
    * I noticed that NegativeLogLikelyhood can be used in cases like this, but it is defined only on inerval (0,1> which is not the case of distribution deviance
    * I noticed some posts about quasilikelikehood, eg. this [post](https://stats.stackexchange.com/questions/225825/what-log-likelihood-function-do-you-use-when-doing-a-poisson-regression-with-con) also probably same idea implemented in the discussion above with abbreviation QLL
    * how do you do it? do you somehow scale the distribution deviance to (0,1>?
    * some additional resources:
    * poison loss calling itself in [pytorch?](https://github.com/pytorch/pytorch/blob/b10929a14a32c90ac5943cc3f27a3a9105e6f53f/torch/nn/functional.py#L2644)
    * impossible to figure out for me how they do it in [h20](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/algo-params/tweedie_link_power.html)
    * [might help?](https://github.com/madrury/py-glm/blob/master/glm/families.py)
**ZILNloss**
* after playing around with it I am not sure if this is really helpful, it seems to me that the idea is plausible but it is almost impossible for me to understand the proof in the original paper as:
  * the authors are using some 0.68 number in KDD dataset that I could not figure out
  * the metrics are hard to understand - they are separated into deciles etc.
* nevertheless, I implemented the authors Keras code in pytorch, including their test and compared the original and pywtorch_widedeep implementation results in the [notebook]()
**QuantileLoss**
* I decided to add new method - "multiregression", as the quantileloss function is regressing multiple values at the same time
  * implementation is "shamelessly" copied from [pytorch-forecasting](https://pytorch-forecasting.readthedocs.io/en/latest/api/pytorch_forecasting.metrics.QuantileLoss.html)
  * @jrzaurin could you please recheck it? it is passing the unittests but it seems to have "strange" values, that are not adhering to idea that the quantile values should be increasing in each sample - maybe just a lack of training etc.

**FINALL UPDATE**:
* branch is ready to be merged, and I can figure out later how to add whole family of tweedie distributions, ie. their losses, to the library

**Comment by 5uperpalo** [comment]
additional notes for TweedieLoss:
* this [post](https://towardsdatascience.com/tweedie-loss-function-for-right-skewed-data-2c5ca470678f) only briefly summarizes this [paper](https://arxiv.org/pdf/1912.12356.pdf) which calculates Probability Density Function and then expresses Negative Log Likelihood, which is a common function used for Losses in DeepNets thanks to it's nice characteristics in interval (0,1)
* this [post](https://sathesant.medium.com/tweedie-loss-function-395d96883f0b) does, in my opinion, slightly better job in summarizing the information
* use of NLL is usually preferred in DeepNets, but other possibility is to use distribution deviance which is not defined as nicely as NLL on interval (0,1), ie. [sklearn approach](https://scikit-learn.org/stable/modules/model_evaluation.html#mean-tweedie-deviance)
* this [post](https://www.statisticshowto.com/tweedie-distribution/) gives additional summary on some(not all) other tweedie family Probability Density Functions, next steps is to do NLL and we have other Loss functions

**Summary**
* current implementation is correct
